mne/time_frequency/tfr.py

```diff
@@ -689,12 +689,17 @@ def plot(self, picks=None, baseline=None, mode='mean', tmin=None,
         from ..viz.topo import _imshow_tfr
         import matplotlib.pyplot as plt
         times, freqs = self.times.copy(), self.freqs.copy()
-        data = self.data[picks]
+        info = self.info
+        data = self.data
```

Comment:
this change could potentially make the code *much* slower. Consider the case where I have ~400 channels, but am only looking at one or two of them. Now all changes from `_preproc_tfr` (a memcopy plus some arithmetic) will be done to many more channels than necessary. We don't need to fix that here, but you should make a note of the potential speed cost here , and potential savings that could be gained from refactoring the relationship between when `picks` are taken and when `_preproc_tfr` operates.